### PR TITLE
WEB-2063 readTextFromClipboard / writeTextToClipboard

### LINKED ISSUE #149: Clipboard access
Due this [jira task OBVIVO-2111](https://jira.tid.es/browse/OBVIVO-2111), vivo needs to access to the clipboard from android and can't at the moment.

When the web tries to access the clipboard from the app it gets an error message about missing permissions, after some research it seems the best way to do that is to use a javascript bridge and use native code for that as seen in [Clipboard API call throws NotAllowedError without invoking onPermissionRequest()](https://stackoverflow.com/questions/61243646/clipboard-api-call-throws-notallowederror-without-invoking-onpermissionrequest/61546346#61546346).

A possible solution would include a couple of bridge methods:

## Read from the clipboard

This method allows to read text from the clipboard.

### Js method
```javascript
clipboardRead: ({}) => Promise<string>
Request → { type: "CLIPBOARD_READ_TEXT", id: string }
Response → { type: "CLIPBOARD_READ_TEXT", id: string, payload: "Parcela PIX" }
```

Message type:

**CLIPBOARD_READ_TEXT**

### Success response
Response with the text from the clipboard. Empty string if it's empty.

## Copy/Write to the clipboard

This method allows to copy text to the clipboard.

In case it is called when the loading screen is not currently shown, it would have no effect.

### Js method
```javascript
clipboardWrite: ({
    message: string;
}) => Promise<void>
Request → { type: "CLIPBOARD_WRITE_TEXT", id: string, payload: "Parcela PIX" }
Response → { type: "CLIPBOARD_WRITE_TEXT", id: string }
```

Message type:

**CLIPBOARD_WRITE_TEXT**

### Success response
Empty response.

### Error response codes
400 → Bad request: If payload is null or it contains no message to copy.
500 → Internal Error: If an error occurred while copying the text to the clipboard.

This could be added in 24.10

### PR DESCRIPTION
**Pull request by atabel**
closes #149

### CODE REVIEW

File: src/post-message.ts [hunk 1]
```diff
@@ -351,6 +351,16 @@ export type ResponsesFromNativeApp = {
...
     };
+    CLIPBOARD_READ_TEXT: {
+        type: 'CLIPBOARD_READ_TEXT';
+        id: string;
+        payload: string;
+    };
+    CLIPBOARD_WRITE_TEXT: {
+        type: 'CLIPBOARD_WRITE_TEXT';
+        id: string;
+        payload: void;
```
Comment (jeprubio) [inline src/post-message.ts:362]: I think the payload should be the text to write into the clipboard in here.

File: src/__tests__/clipboard-test.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+test('write to clipboard', async () => {
+    createFakeWebKitPostMessage({
+        checkMessage: (msg) => {
+            expect(msg.type).toBe('CLIPBOARD_WRITE_TEXT');
+            expect(msg.payload).toBe('some text');
+        },
+        getResponse: (msg) => ({
+            type: 'CLIPBOARD_WRITE_TEXT',
+            id: msg.id,
```
Comment (jeprubio) [inline src/__tests__/clipboard-test.ts:29]: I think there should be a payload with the text to write in here.

File: src/__tests__/clipboard-test.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import {createFakeWebKitPostMessage} from './fake-post-message';
+
+test('read from clipboard', async () => {
+    createFakeWebKitPostMessage({
+        checkMessage: (msg) => {
+            expect(msg.type).toBe('CLIPBOARD_READ_TEXT');
+        },
+        getResponse: (msg) => ({
+            type: 'CLIPBOARD_READ_TEXT',
+            payload: 'some text',
```
Comment (jeprubio) [inline src/__tests__/clipboard-test.ts:11]: Should read text have a payload?

File: src/post-message.ts [hunk 4]
```diff
@@ -351,6 +351,16 @@ export type ResponsesFromNativeApp = {
             isPowerSafeMode: boolean;
         };
     };
+    CLIPBOARD_READ_TEXT: {
+        type: 'CLIPBOARD_READ_TEXT';
+        id: string;
+        payload: string;
```
Comment (jeprubio) [inline src/post-message.ts:357]: Should read text have a payload?

File: src/__tests__/clipboard-test.ts (hunk 3 above)
Comment (jeprubio) [reply src/__tests__/clipboard-test.ts:11]: Oh, sorry, this is the response, It's fine, my bad

**Review by jeprubio — APPROVED** [review]
👍

**Review by marcoskolodny — APPROVED** [review]

**Review by pladaria — APPROVED** [review]

**Pull request merged**